src/neomaril_codex/training.py

```diff
@@ -455,22 +485,94 @@ def __upload_training(self, run_name:str, train_data:str, training_reference:Opt
                 
                 upload_data += extra_data
                 
-            form_data = {'run_name': run_name, 'training_reference': training_reference,
-                                    'python_version': "Python"+python_version.replace('.', '')}
+            form_data['training_reference']= training_reference,
+            form_data['python_version'] = "Python"+python_version.replace('.', '')
         
         elif self.training_type == 'AutoML':
                 
-            form_data = {'run_name': run_name}
-
             if conf_dict:
-                if isinstance(conf_dict, str):
-                    schema_file = open(conf_dict, 'rb')
-                elif isinstance(conf_dict, dict):
-                    schema_file = io.StringIO()
-                    json.dump(conf_dict, schema_file).seek(0)
-                upload_data.append(("conf_dict", ("conf.json", schema_file)))
+                upload_data.append(("conf_dict", ("conf.json", parse_dict_or_file(conf_dict))))
             else:
                 raise InputError("conf_dict is mandatory for AutoML training")
+            
+        elif self.training_type == 'External':
+            upload_data = []
+
+            with TemporaryDirectory() as temp_dir:
+
+                #parquets
+                features_path = os.path.join(temp_dir, 'features.parquet')
+                target_path = os.path.join(temp_dir, 'target.parquet')
+                predictions_path = os.path.join(temp_dir, 'predictions.parquet')
+
+                X_train.to_parquet(features_path)
+                y_train.to_parquet(target_path)
+                model_outputs.to_parquet(predictions_path)
+
+                upload_data = [
+                    ("features", ('features.parquet', open(features_path, 'rb'))),
+                    ("target", ('target.parquet', open(target_path, 'rb'))),
+                    ("output", ('predictions.parquet', open(predictions_path, 'rb'))),
+                ]
+
+                # model 
+                model_path = os.path.join(temp_dir, 'model.pkl')
+                
+                with open(model_path, "wb") as f:
+                    pickle.dump(model_file, f)
+
+                upload_data.append(
+                    ("model",
+                    ('model.pkl', open(model_path, 'rb')))
+                )
+
+                # parameters 
+                params_path = os.path.join(temp_dir, 'params.json')
+                with open(params_path, "w") as f:
```

Comment:
I would like to find more params like we do [here](https://github.com/datarisk-io/neomaril/blob/develop/worker/neomaril/src/neomaril/training/executor.py#L87-L143)

Reply:
Solved!

Reply:
Solved!